pilot/pkg/networking/networking.go

```diff
@@ -93,6 +95,11 @@ type FilterChain struct {
 	// ListenerProtocol indicates whether this filter chain is for HTTP or TCP
 	// Note that HTTP filter chains can also have network filters
 	ListenerProtocol ListenerProtocol
+	// GatewayServerTLSMode is set only when this filter chain is part of a Gateway, and
+	// the Server corresponding to this filter chain is doing TLS termination with ISTIO_MUTUAL as the TLS mode.
+	// This allows the authN plugin to add the istio_authn filter to gateways in addition to sidecars.
+	GatewayServerTLSMode networking.ServerTLSSettings_TLSmode
```

Comment:
Can we use a simple bool flag `istioMutualGateway` if it is only supposed to be set and used with ISTIO_MUTUAL mode? This means other modes will not set this field then it makes the enum needless and could cause confusion when the field is used in other places.